include/yaramod/types/token_stream.h

```diff
@@ -123,6 +123,12 @@ class TokenStream
 	std::vector<std::string> getTokensAsText() const;
 	/// @}
 
+	/// @name New Line Characters
+	/// @{
+	const std::string& getNewLineStyle() const { return new_line_style; }
+	void setNewLineChar(std::string&& line) { new_line_style = line; }
```

Comment:
Beware that this copies `line` into `new_line_style`, i.e. it calls the copy-assignment operator. If you want it to get moved, you need to `std::move(line)` there:
```diff
-void setNewLineChar(std::string&& line) { new_line_style = line; }
+void setNewLineChar(std::string&& line) { new_line_style = std::move(line); }
```
The reason is that even though `line` is passed as an r-value to `setNewLineChar()`, inside the body, it is actually an l-value (it has a name: `line`). Thus, to make it an r-value, we need `std::move()`.

To simplify the implementation (and stop forcing callers of the function to pass an r-value), consider passing the parameter simply by value and move it into `new_line_style`:
```diff
-void setNewLineChar(std::string&& line) { new_line_style = line; }
+void setNewLineChar(std::string line) { new_line_style = std::move(line); }
```

Reply:
Thank you very much! The second option seems to be better here, so I will pass by value.